File: src/main/java/com/vjiki/music/dto/CommentResponse.java
package com.vjiki.music.dto;

import java.time.OffsetDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

import com.fasterxml.jackson.annotation.JsonProperty;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class CommentResponse {
    private UUID id;

    @JsonProperty("track_id")
    private UUID trackId;

    @JsonProperty("user_id")
    private UUID userId;

    @JsonProperty("user_nickname")
    private String userNickname;

    @JsonProperty("user_avatar_url")
    private String userAvatarUrl;

    @JsonProperty("parent_id")
    private UUID parentId;

    private String content;
    private String status;

    @JsonProperty("likes_count")
    private Integer likesCount;

    @JsonProperty("replies_count")
    private Integer repliesCount;

    @JsonProperty("is_liked")
    @Builder.Default
    private Boolean isLiked = false;

    @JsonProperty("created_at")
    private OffsetDateTime createdAt;

    @JsonProperty("updated_at")
    private OffsetDateTime updatedAt;

    @Builder.Default
    private List<CommentResponse> replies = new ArrayList<>();
}

File: src/main/kotlin/com/vjiki/music/dto/CommentResponse.kt
package com.vjiki.music.dto

import com.fasterxml.jackson.annotation.JsonProperty
import java.time.OffsetDateTime
import java.util.UUID

data class CommentResponse(
    val id: UUID,
    @JsonProperty("track_id")
    val trackId: UUID,
    @JsonProperty("user_id")
    val userId: UUID,
    @JsonProperty("user_nickname")
    val userNickname: String?,
    @JsonProperty("user_avatar_url")
    val userAvatarUrl: String?,
    @JsonProperty("parent_id")
    val parentId: UUID?,
    val content: String,
    val status: String,
    @JsonProperty("likes_count")
    val likesCount: Int,
    @JsonProperty("replies_count")
    val repliesCount: Int,
    @JsonProperty("is_liked")
    val isLiked: Boolean = false,
    @JsonProperty("created_at")
    val createdAt: OffsetDateTime?,
    @JsonProperty("updated_at")
    val updatedAt: OffsetDateTime?,
    val replies: List<CommentResponse> = emptyList()
)
